site/content/extend/plugins/best-practices.md

```diff
@@ -0,0 +1,33 @@
+---
+title: "Example Plugins"
+subsection: Plugins (Beta)
+weight: 50
+---
+
+See here for [server-specific best practices for plugins](/extend/plugins/server/best-practices/). Webapp-specific best practices are incoming.
+
+## How can I review the entire code base of a plugin?
+
+Carry out the following steps:
+
+1. Take a backup of project directory
+2. Create a dummy-master branch with no code:
```

Comment:
What I did was commit the code that I didn't want to review. Like, you probably don't want to review boilerplate files and boilerplate code.
So after the `git rm -rf` you could have `git add ....` where `....` are the files that you don't want to review.
then `git commit -m "Files we don't want to review"`
Something like that.

Reply:
@cpoile How's this?

----

## How can I review the entire code base of a plugin?

Carry out the following steps:

1. Take a backup of project directory
2. Create a dummy-master branch with no code:

   ```
   git checkout --orphan dummy-master
   git rm -rf
   ```

3. Commit code that you do not want to review:

   ```
   git add <files excluded from review>
   git push origin dummy-master
   ```
   
4. Create a dummy-review branch from dummy-master:

   ```
   git checkout -b dummy-review
   git add
   git commit -m "Files to review"
   git push origin dummy-review
   ```
   
5. Create a PR from dummy-review -> dummy-master

6. Code review on the resulting PR

Reply:
@jasonblais Awesome!

Reply:
@jasonblais oh wait, you might want to gut a commit in step 3

Reply:
@cpoile Updated 👍